qiita_pet/handlers/api_proxy/prep_template.py

```diff
@@ -6,52 +6,358 @@
+    user_def_type : str
+        The user-defined investigation type
+    new_type : str
+        The new user-defined investigation_type
+
+    Returns
+    -------
+    str
+        The investigation type chosen by the user
+    """
+    if inv_type == '':
+        inv_type = None
+    elif inv_type == 'Other' and user_def_type == 'New Type':
+        # This is a nre user defined investigation type so store it
+        inv_type = new_type
+        ontology = Ontology(convert_to_id('ENA', 'ontology'))
+        ontology.add_user_defined_term(inv_type)
+    elif inv_type == 'Other' and user_def_type != 'New Type':
+        inv_type = user_def_type
+    return inv_type
+
+
+def prep_template_get_req(prep_id, user_id):
+    """Gets the json of the full prep template
+
+    Parameters
+    ----------
+    prep_id : int
+        PrepTemplate id to get info for
+    user_id : str
+        User requesting the sample template info
+
+    Returns
+    -------
+    dict of dictionaries
+        Dictionary object where the keys are the metadata samples
+        and the values are a dictionary of column and value.
+        Format {sample: {column: value, ...}, ...}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    df = prep.to_dataframe()
+    return {'status': 'success',
+            'message': '',
+            'template': df.to_dict(orient='index')}
+
+
+def prep_template_summary_get_req(prep_id, user_id):
+    """Get the summarized prep template data for each metadata column
+
+    Parameters
+    ----------
+    prep_id : int
+        PrepTemplate id to get info for
     user_id : str
-        User id requesting the prep templates
+        User requesting the sample template info
 
     Returns
     -------
-    dict
-        Prep template information in the form
-        {'status': status,
-         'message': msg,
-         'info': dict of list of dict}
-        status can be success, warning, or error depending on result
-        message has the warnings or errors
-        info contains prep template information seperated by data type, in the
-        form {data_type: [{prep 1 info dict}, ....], ...}
+    dict of objects
+        Dictionary object where the keys are the metadata categories
+        and the values are list of tuples. Each tuple is an observed value in
+        the category and the number of times its seen.
+        Format {status: status,
+                message: message,
+                num_samples: value,
+                category: [(val1, count1), (val2, count2), ...], ...}
+    """
+    prep = PrepTemplate(int(prep_id))
+    access_error = check_access(prep.study_id, user_id)
+    if access_error:
+        return access_error
+    df = prep.to_dataframe()
+    out = {'num_samples': df.shape[0],
+           'summary': {}}
+
+    # drop the prep_id column if it exists
+    if 'study_id' in df.columns:
```

Comment:
I don't think this type of handling should be done outside of the prep template. Why are you checking if the study_id is present? Isn't it showing a knowledge of how that data is stored outside of the DB?

Also, recent discussions showed that we may want to add the study_id as a column, since it will help for meta-analysis - but that is a different discussion.